tests/phpunit/api/v3/PaymentTest.php

```diff
@@ -406,4 +406,61 @@ public function testUpdatePayment() {
     ));
   }
 
+  /**
+   * Test create payment api for paylater contribution
+   */
+  public function testCreatePaymentPayLater() {
+    $contributionParams = array(
+      'total_amount' => 100,
+      'currency' => 'USD',
+      'contact_id' => $this->_individualId,
+      'financial_type_id' => 1,
+      'contribution_status_id' => 2,
+      'is_pay_later' => 1,
+    );
+    $contribution = $this->callAPISuccess('Contribution', 'create', $contributionParams);
+
+    //Create partial payment
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 100,
+    );
+    $payment = $this->callAPIAndDocument('Payment', 'create', $params, __FUNCTION__, __FILE__);
```

Comment:
From my perspective I'd like a test suite that exercises as many code paths as possible. Sometimes the properties of an object passed as a param may change even though the form of the param being passed is the same; not sure if that is case here, as I don't understand why lines 421-428 are repeated both above and below. The problem I have with 423-428 is that the comment says partial payment but it is a full payment on a pay later.